utilities.js

```diff
@@ -0,0 +1,148 @@
+module.exports = isEqual
```

Comment:
the signature of this winds up violating a little bit of principle of least surprise for me

so when you consume this method, the code looks like this (from above):

```js
let isEqual = require("../utilities")
```

But it's weird to have a function name called `isEqual` as the result of pulling in `"utilities"`.  It likely the case that the motivation was to keep most functions private and not export them, but it might be better to export as an object with the function as a key:

```js
module.exports = { isEqual }
```

Then when consuming, you could either destructure right away because the only thing worth pulling in will be the fn:

```js
let { isEqual } = require("../utilities")
isEqual(...)
```

Or just pull in the lib and use the fn from the object:

```js
let utils = require("../utilities")
utils.isEqual(...)
```